qemu/tests/blockdev_stream_base_itself.py

```diff
@@ -0,0 +1,111 @@
+import logging
+
+from virttest import data_dir
+from virttest.qemu_monitor import QMPCmdError
+
+from provider import backup_utils
+from provider.blockdev_stream_base import BlockDevStreamTest
+from provider.virt_storage.storage_admin import sp_admin
+
+
+class BlockdevStreamBaseitself(BlockDevStreamTest):
+    """Do block-stream based on itself"""
+
+    def __init__(self, test, params, env):
+        super(BlockdevStreamBaseitself, self).__init__(test, params, env)
+        self._snapshot_images = self.params.objects("snapshot_images")
+        self._trash = []
+
+    def _disk_define_by_params(self, tag):
+        params = self.params.copy()
+        params.setdefault("target_path", data_dir.get_data_dir())
+        return sp_admin.volume_define_by_params(tag, params)
+
+    def prepare_snapshot_file(self):
+        """hotplug all snapshot images"""
+        for tag in self._snapshot_images:
+            disk = self._disk_define_by_params(tag)
+            disk.hotplug(self.main_vm)
+            self._trash.append(disk)
+
+    def snapshot_test(self):
+        """create one snapshot, create one new file"""
+        self.generate_tempfile(self.disks_info[self.base_tag][1],
+                               filename="base",
+                               size=self.params["tempfile_size"])
+
+        self.snapshot_chain = [self.base_tag] + self._snapshot_images
+        for idx in range(1, len(self.snapshot_chain)):
+            backup_utils.blockdev_snapshot(
+                self.main_vm,
+                "drive_%s" % self.snapshot_chain[idx-1],
+                "drive_%s" % self.snapshot_chain[idx]
+            )
+
+            self.generate_tempfile(self.disks_info[self.base_tag][1],
+                                   filename=self.snapshot_chain[idx],
+                                   size=self.params["tempfile_size"])
+
+    def _remove_images(self):
+        for img in self._trash:
+            sp_admin.remove_volume(img)
+
+    def post_test(self):
+        try:
+            if self.clone_vm.is_alive():
+                self.clone_vm.destroy()
+            #self._remove_images()
```

Comment:
_remove_images is defined but never used?

Reply:
Have updated, thanks zhenchao.